### ISSUE TITLE
Maybe.fromEmpty?

### ISSUE THREAD

**Issue by customcommander** [monet/monet.js#213]
I was wondering if it would make sense to have this new static method?

```javascript
Maybe.fromEmpty([]);
//=> None
Maybe.fromEmpty({});
//=> None
Maybe.fromEmpty('');
//=> None
Maybe.fromEmpty(null);
//=> None
Maybe.fromEmpty(undefined);
//=> None
```

It could (or perhaps should) delegate to [`fantasy-land/empty`](https://github.com/fantasyland/fantasy-land#fantasy-landempty-method) to work out if a given value is its type's empty value.

Use case:

I have a `css` rules object which contains css rules to apply to a string if there's any. Otherwise the string is returned as is. Problem is that the object may actually be "empty" i.e. it is an object but has no properties.

```javascript
const styles = (css, text) =>
  Maybe
    .fromEmpty(css) // e.g. {'text-decoration': 'underline', 'color': 'red'}
    .map(obj => obj.entries()) // [['text-decoration', 'underline'], ['color', 'red']]
    .map(arr => arr.map(pair => pair.join(':'))) // ['text-decoration:underline', 'color:red']
    .map(arr => arr.join(';')) // 'text-decoration:underline;color:red'
    .fold(text)(str => `<span style="${str}">${text}</span>`);

styles({'font-style': 'italic'}, 'foo');
//=> '<span style="font-style:italic">foo</span>'

styles({}, 'foo');
//=> 'foo'
```

Any thoughts?

**Comment by ulfryk** [comment]
👍 great idea @customcommander 
I had one usecase just today :)

Can you implement it?

**Comment by customcommander** [comment]
I would love it. I'll give it a shot :)

**Comment by ulfryk** [comment]
snapshot released as monet@0.9.0-422
